api/src/main/scala/dx/api/DxInstanceType.scala

```diff
@@ -524,29 +537,28 @@ object InstanceTypeDB extends DefaultJsonProtocol {
             )
         )
     )
-    val (availableInstanceTypes, pricingAvailable) =
-      try {
-        val pricingModel = getPricingModel(api, projectDesc.billTo.get, projectDesc.region.get)
-        val instanceTypesWithPrices = allInstanceTypes.keySet
-          .intersect(pricingModel.keySet)
-          .map { name =>
-            name -> allInstanceTypes(name).copy(price = pricingModel.get(name))
-          }
-          .toMap
-        (instanceTypesWithPrices, true)
-      } catch {
-        case ex: Throwable =>
-          logger.warning(
-              """|Warning: insufficient permissions to retrieve the
-                 |instance price list. This will result in suboptimal machine choices,
-                 |incurring higher costs when running workflows.
-                 |""".stripMargin.replaceAll("\n", " ")
-          )
-          (allInstanceTypes, false)
+    val availableInstanceTypes =
+      getPricingModel(api, projectDesc.billTo.get, projectDesc.region.get)
+        .map { pricingModel =>
+          allInstanceTypes.keySet
+            .intersect(pricingModel.keySet)
+            .map { name =>
+              name -> allInstanceTypes(name).copy(price = pricingModel.get(name))
+            }
+            .toMap
+        }
+    val usableInstanceTypes = availableInstanceTypes
+      .getOrElse {
+        logger.warning(
+            """|Warning: insufficient permissions to retrieve the instance price list.
```

Comment:
So now it looks like we log this only if the PermissionDeniedException happens, ok. What happens next if the map was empty, and therefore usableInstanceTypes is empty? It shouldn't happen, but how is that case handled?

Reply:
the message now happens if the pricing model isn't available for any reason.